### ISSUE TITLE
Use sparse set components for fragmented_iter for Bevy

### ISSUE THREAD

**Issue by alice-i-cecile** [TomGillen/ecs_bench_suite#25]
As discussed in the [0.5 news release](https://bevyengine.org/news/bevy-0-5/#hybrid-component-storage-the-solution), sparse set component storage was designed to make fragmented iteration faster, precisely the use case being tested in the `fragmented_iter` benchmark.

**Comment by cart** [comment]
Yup other ECS-es currently use "fast path" apis, such as SoA insertion and for_each iterators (despite other apis being "more idiomatic" or "recommended"). I think it is fair play at this point to do the same for Bevy.

However this is my current take on ecs_bench_suite policy/labeling in general: https://github.com/rust-gamedev/ecs_bench_suite/pull/18#issuecomment-811553276

The current approach biases toward the wrong behaviors and makes it harder for developers to make an informed decision when picking an ECS.
